### ISSUE TITLE
Xeno-thoughts

### ISSUE THREAD

**Issue by frameworld** [endless-sky/endless-sky#577]
–––0––– Text, spelling/words
Is the alien-language right now just another human language (english, ru, ita, ger, ...)?
 * Shared-unknown/Unique?

The game could have an alien-language dictionary.
 * Each species an alphabet which contains phonetic symbols. (aeiou, kcx).
  *  auto-generated by a letters and expanded by similarities to a fixed number of n symbols.
 * Each english word can be translated to that alphabet.
  *  use an initial seed of random translations for words needing roughly the same length.
  *  each syllable can contain only one of a letter from which some alias others ( t→d )

It would be an interesting challenge to decode this.

–––1––– Language patterns
A question can demand [silent acceptance or forced denial] ( "~, or not?" )
A question can state [the fact that you are not sure about something] (A question that doesn't demand an answer by context )

A statement can be a question ( "You said X=true. Y=X. Y=true then, if I didn't miss-understood you." )​

A language can specify first the later or specify later the first.

It can have references to the last or next sentence, to a pattern elsewhere (usually a repeated word or it, this, these referring to a subject).

–––2––– Editor, AI
AI-ideas:
 * Find a symbol-pattern (something that repeats).
 * Find a mask-pattern (a x b)→{[function→code: (x: a, b)], [alias→math: (a * b)]} with variable inputs/outputs.
 * Find definitions (example: function+code→"./code/function/x, f(a, b)", alias+math→"./math/alias/x"
 * Find new concepts of thinking (search-routes/strategies for definitions that apply to a symbol or pattern)

I spent much time to define the process of thinking in math and program-code, it's still a very unstable concept.

Every 3 days on average I would have to overhaul the complete system for a new optimisation, sub-concept, firewall+security check, …. It is faster to draft the concept in my mind than writing it down.

It is reliable enough because it is harmonic/integral knowledge unlike things you just need to remind in a day/week, is subject to invalidation in future or a thing about past that doesn't repeat in a pattern you recognize.
 * It is easy to remember things that fit these criteria and which you are interested into.

If somebody found a work-group for this topic, please give me a hint.

**Comment by bobafetthotmail** [comment]
Hmmm... is this the backend of some kind of minigame to understand what an alien species is trying to say?

Now that's something unexpected.

We have already Kroath that could use something like that, as their first (and only) mission is a first contact, and they don't speak english (or the player's language anyway). They say random stuff (afaik), and both them and the player resort to drawing to get the concepts across.

>"It is faster to draft the concept in my mind than writing it down"

You tried to write it down as pseudocode instead? It's closer to pure logic reasoning than a programming language (or math for that matter), and is useful to put down the logic of what you need to do without caring at all about all the syntax of the average programming language. 
some examples http://www.unf.edu/~broggio/cop2221/2221pseu.htm

Although I'm suspecting that on something you are aiming a bit too high, in the realm of actual AI programs working with language like say Cortana/Google Now.

>"If somebody found a work-group for this topic, please give me a hint."

It is some kind of cryptography, a weak kind (as it has to be somewhat translatable), being applied on either grammar or syntax.

Might want to read up on old and/or obsolete cryptographic algorithms, those used before it became pure incomprehensible math.

Or just look up other attempts at alien alphabets done in fiction, apart from the "new font from the same letters" ones (duh!) there are quite a few interesting concepts http://io9.gizmodo.com/5218119/13-alien-languages-you-can-actually-read
Also fantasy languages.

**Comment by frameworld** [comment]
Yes, even pseudo-code takes a lot of time.

Also I have problems writing long texts not on a computer and I like to keep the computer off while resting or reading a book.

> It is some kind of cryptography
> Or just look up other attempts at alien alphabets done in fiction

I think of it as a form of information formatting and compression which is a bit different from cryptography on arbitrary strings.
Fantasy and Sci-Fi languages look interesting, but important for my interest is that they are processable by computers.

Have you seen the wikipedia page on https://en.wikipedia.org/wiki/Newspeak from Nineteen-Eighty-Four, a roman from earlier that century? I've seen it and try the opposite with C++/Java/…

For translations, you need to tell a computer about the ambiguity of symbols and perhaps use a pathfinder through symbol-nodes rather than way-points + some more rules we don't think much about IRL when we are older than 3 :)

**Comment by bobafetthotmail** [comment]
>"For translations, you need to tell a computer about the ambiguity of symbols and perhaps use a pathfinder through symbol-nodes rather than way-points + some more rules"

Ok, my suspicions are confirmed. That's a field called "natural language processing", the realm of modern bleeding-edge AI programs that deal with language. A quick google search to know more.

It is waaay above my understanding, and probably a bit too overkill to get in a game unless you manage to dumb it down a lot.

>" we don't think much about IRL when we are older than 3 :)"
Human beings have "hardware acceleration" for language, there are specific parts of the brain that do that on their own, you learn how to use them by trial and error, it's not rational rules you "know".
Like walking, most of it is automated by reflexes, you don't "know" how to walk, you "learn" how to walk.

That's why the AI field is hard. You need to figure out first how you do stuff before you can write a program to do the same automatically.

**Comment by frameworld** [comment]
Bobafetthotmail, Your second quote isn't separated by a paragraph.

> Human beings have "hardware acceleration" for language, there are specific parts of the brain that do that on their own, you learn how to use them by trial and error, it's not rational rules you "know".

I don't know if I understand "hardware acceleration" correctly, please give me a hint into where it fits.
See below for a bit of context related for hardware→subconsciousness.

––– subconsciousness→origin/expected
There is a gap in memory to your early self before 3-4 years. I can barely remember the vacation in Greek when I was 3 years and 1-4 month and heard it is common.
Perhaps this gap is responsible for our sub-consciousness (along other gaps like this most recall-able one).

––– subconsciousness→my-experience
Maybe I've learned (or re-learned from another perspective) most applications of basic language syntax after that age since over 18 years now, by trying to act conform with the expectations in school.

I always know at least a few most-significant causes which are essential to my current primary thoughts.

One definition of thinking is following the most interesting thought-path. Interest is defined by subjectively assigned need (where that need comes from does not matter - grabbing food, learn for later life, … -it depends on expected gains).

––– subconsciousness→difference-to-consciousness/expected
Often I think one way and the other way and notice the gap between both ways when they merge and the result reminds about parts of both.
 * When looking back a half second I remember both ways to be thought - and double speed of total thinking.
 * Every time I notice a change - in what I remember or a motion of my body I notice the thought-path that led to it up to another change / remind-able thought-path.

From the effects I observe, I'd say that what we refer to with "thinking" is actually the decision whether a thought will have a permanent effect on us or the reality.
 * Thinking is [0] deciding about the progress.
 * Thinking is [1] using your subconsciousness.

––– communication,difference→difference-observed-to-others
And (not only) while writing this I have notice every time I type that I am considering alternative view-points of what I have written -fully aware, not subconscious.
I consider a dozen ways of writing about each aspect(associated thought) sentence in a split-second and notice the difficulty to create a linearized version of my thoughts.
Every word is subject to consideration as fast as I can type (I type about 4-5 letters a second, 3x as many when writing stuff for myself) - every word, a reconsideration of the sentence.


Writing is so slow. You have to translate meanings between words into words and make sure nobody takes the wrong direction - even if you don't see the response of your reading-opponent.

**Comment by Pointedstick** [comment]
Can someone explain this request in English?

**Comment by frameworld** [comment]
Tried. Now it's your turn to think ;)

**Comment by Nightmareman** [comment]
@Pointedstick Don't feel bad almost everyone has problems understanding him the structure of the sentences and word usage are both very strange. I think I know what he is trying to say but not 100% sure.

It appears he is talking about a mini-game to translate alien text so you can talk to all the alien races we will likely meet in the future and the Korath as well. Think it would take the form of a dialog tree with there being correct and incorrect answers if I am understanding what he is saying or perhaps a different type of mini-game entirely.

@frameworld With some of your sentences the structure is so bad that it is nearly impossible to decipher what you say so don't respond with something like. "It's your turn to think." Use better English then we wouldn't have to spend 2 hours trying to figure out what you are saying.

**Comment by bobafetthotmail** [comment]
>"I don't know if I understand "hardware acceleration" correctly, please give me a hint into where it fits."

The part of the brain where "thinking" happens is the pre-frontal cortex (this is obvious with imaging techniques), the rest of the brain isn't related to consciousness but dedicated to various different functions. Managing the body, emotions, sensor feed elaboration, various kinds of memory, motor control and tons of motor automation to do the leg work.
It's pretty well-proven, look up "neuromotor programming".


Still, please state the goal of this. I asked already and got no answer.
How does this relate to the game?

**Comment by sanjayatio** [comment]
cmiiw: frameworld suggests:
- a dictionary to map english with some fancy alphabeths (wingdings)?
- an engine to translate english sentences into any alien languages based on the mapping above.

if these can be executed properly, it will be beneficial for content creators (by unloading the burden to invent a coherent alien language).
but...

it seems to me that the discussion has strayed to an un (or not so) related path.
i'm of the same mind with bobafetthotmail - it's an overkill for this game.
it's almost impossible to understand a language by translating it word for word,
except when the languages share the exact same structure.
- try to use google translate to process japanese/korean language .
- some dedicated "machine translators" chop a sentence in many different ways, feed them into the translator
and to find the combination that make the most sense (in the destination language and in the context of said sentence).

imho this thought process is more of an analog process rather than a digital yes/no one.
worse, just as frameworld said it happens on the subconscious level - not everyone can give detailed explanation (breakdown) of his/her own thought process.
sure, you might be able to make a model for it (with effort) but...


might be easier to use another human language (just as what you asked in the 1st sentence).
look at this: https://en.wikipedia.org/wiki/Javanese_language#Registers
it descended from (indian) sanskrit but has evolved into its own. 
would be interesting if different aliens use different styles:
- drak uses krama - the most polite and formal one
- quarg/pug uses madya - the middle one
- hai/korath uses ngoko - the crude one
disclaimer: not actually suggesting the bullet points above; those just meant to illustrate my point.
i can only speak ngoko. the javanese alphabeths have been incorporated into the unicode btw.

perhaps some linguist can try to apply old celtic/german/english instead &/ use the "druidic" alphabeths used in some rpg games.

**Comment by bobafetthotmail** [comment]
>"not everyone can give detailed explanation (breakdown) of his/her own thought process."

Translating from a language to another requires understanding the meaning in language A, then finding a way to convey the same meaning in language B.

It always means basically rewriting the whole text from scratch, and it NEVER involves any kind of weird trick as above.

There is no math, permutations, patterns, whatever. Read, extract the meaning, write from scratch in another language.

Human languages are arbitrary binary associations between words and meanings, grammar is arbitrary, syntax is arbitrary, and also sayings are arbitrary, conventional logic will always fail horribly because a human language it's not anywhere near logic.

To get a machine that can comprehend a human language they are developing machine learning and artificial intelligence, with neural networks, and all that jazz. It must mean something.

If the necessity is just making a dictionary to help people making an alien language, write a wiki page where you make a bunch of examples on how they can put down some arbitrary rules to put a meaning into a word, stealing ideas from real-world languages and fictional ones.

Say, I decide that "ship" is said "waisk".
I also decide that in my language it's the singular form of the noun that needs a modifier to the word, and the modifier is 'kven.

So "ship" is "waisk'kven" and "ships" is "waisk".

And so on and so forth. Boom, I've made a language!
Is it so hard?

You can also make total masochist languages with ideograms like mandarin/japanese where each thing has a meaning alone and has or might have a different meaning if close to another thing, like the Tarot cards.

For non-ideogram based languages can also add a custom alphabet on top, but an alphabet is just a phonetic translation table (again arbitrary) added on top of the sounds you have associated to the meaning.

If I decide that the sound "wai" is codified by the symbol YY, and I decide that the sound "sk" is codified by the symbol W, when my aliens read "YYW" they say "waisk" with their voice and they mean "ships" in english.

You can go on and make words for concepts that are difficult to translate directly, or don't have a word in english.

It's all beautifully arbitrary, if you need new stuff you come up with it and go on, and the result will even look realistic.

**Comment by Pointedstick** [comment]
This is a very enlightening discussion on linguistics… but what does it have to do with a space opera computer game?

**Comment by bobafetthotmail** [comment]
Having all aliens speak english eventually starts to look dumb with or without magic gimmicks ala Star Trek. 

Giving them a language also allows "first contact" type missions where if you manage to understand their gibberish you can actually make better choices or progress better in their mission tree.

Did you see the Kroath mission? That's a first contact, the Kroath you interact with don't speak in english.
It is a stub so it is short, and has no effect on the game as a whole, but it's interesting.

**Comment by sanjayatio** [comment]
> ... masochist languages 

indeed.

maybe it doesn't have to be a full-blown (read: proper) new alien language.
for the pilot, start with something as simple as alphabets substitution?
- use @%$!*~ for the vowels to accentuate the alien feel?
- make translator machine as a special outfit.
- without such outfit, the game should obscure simple english into an alien language

if this goes well, we can slowly expand/overload the implementation into word dictionaries (i fear it will blow the game size) and those fancy grammar/structure - sentence construction and deconstruction.

**Comment by natangold** [comment]
A translation outfit sounds nifty as hell - and would make an awesome plot-line. Imagine traveling toward the eastern rim of the galaxy, unable to communicate with anything until you run into some super-advanced empath-species with a translator, then traveling back toward human space brokering peace (or war, whatevs) among the races as you go.

Rather than implement one or more alien languages, though, it would probably hella simpler to use a block cipher and give each alien species a unique key that could be unlocked by the player and, through story-lines, given to other factions as well. Because I'm sure the unfettered Hai would just love to be able to communicate with the Korath.

**Comment by frameworld** [comment]
––– Summary : for lazy readers –––

I've learned that real-linguistics and game-linguistics are different topics.

In game, you want basically 3 things:
 * word-word translators.
 * respecting of certain rules of one language (singular, nomen, verb) see Wikipedia:newspeak.
 * Yoda-speak requires above "Yoda: Good, this idea is"→"Human: This idea is good."

Each species could have 2-3 different Yoda-Rules (idea comes from "Yoda", the character of StarWars).
 * This is easy to implement and unique.
 * Once enough rules are done, aliens could easily use some selected 5 out of 25 total to be unique.

For word-word translations, have a look at
 * the zip-compression algorithm.
 * listings of syllables in different languages.

For save-files:
Language Tharok
> (a) is (b) : (b) is (a)
> is : ak
> [is] equal : ak'arath

Player-made rules and achievements can merge in the "Language indentation".
Important things:
 * Order of rules
 * Correctness of rules
 * Human→Alienese (top-down), Alienese→Human (bottom-up)

That's the basic idea.


––– Quotes : answering new posts in detail –––

> It appears he is talking about a mini-game to translate alien text

Optional. Some Easter-Eggs may require it, not for required mission though.

> so you can talk to all the alien races we will likely meet in the future and the Korath as well. Think it would take the form of a dialog tree with there being correct and incorrect answers if I am understanding what he is saying or perhaps a different type of mini-game entirely.

Each correct answer gives an achievement.
You could enter own equations which would give achievements if they are correct, as well, but that would only 1. save time or 2. be required for optional missions.

@frameworld With some of your sentences the structure is so bad that it is nearly impossible to decipher what you say so don't respond with something like. "It's your turn to think." Use better English then we wouldn't have to spend 2 hours trying to figure out what you are saying.

I call it specific, not bad.
The post could be as long as a big wiki-article. I made a summary and hid the other information between the lines.

"It's your turn to think" is not the whole message. You forgot "(I )tried.". I meant, I require further feedback to be able to improve my sentence structure, without feedback I can't make better.

Don't forget this thread is called "Xeno-thoughts" Xeno=Alien ;) - Do not join some UFO-sect now! ;)

> Human beings have "hardware acceleration" for language, there are specific parts of the brain that do that on their own, you learn how to use them by trial and error, it's not rational rules you "know".
Like walking, most of it is automated by reflexes, you don't "know" how to walk, you "learn" how to walk.

> The part of the brain where "thinking" happens is the pre-frontal cortex (this is obvious with imaging techniques), the rest of the brain isn't related to consciousness but dedicated to various different functions. Managing the body, emotions, sensor feed elaboration, various kinds of memory, motor control and tons of motor automation to do the leg work.
It's pretty well-proven, look up "neuromotor programming".

Your answer avoids a reference to the original context ('hw acc' for language | no rational rules you "know").
And I "know". Creating language is rational thinking for me. I can't understand why you don't "know" it.

Any weirdness of sentence structure is part of [what can be described by "The Game Theory"].
I observe language is shortened by putting the part that tells "this is uncertain if" between the lines. There is a lot "reading between the lines".

> a dictionary to map english with some fancy alphabeths (wingdings)?
> an engine to translate english sentences into any alien languages based on the mapping above.

Wingdings give no hint about the spelling. While talking to aliens, spelling is important.
They might say ["!-$%!&" in their language] but you understand "I'szib" which is still alphanumeric.

"Yoda: I speak to you" _can_ become "Yoda: to you, I speak" and "Yoda: ot uyo, I spak". Just one example.

> if these can be executed properly, it will be beneficial for content creators (by unloading the burden to invent a coherent alien language).
but...
> it seems to me that the discussion has strayed to an un (or not so) related path.
i'm of the same mind with bobafetthotmail - it's an overkill for this game.
> it's almost impossible to understand a language by translating it word for word,
except when the languages share the exact same structure.

But it's easy to create non-understandable language by a loss-less back-forth translation algorithm. Obscuring, similar to compression, but without reducing size.

And if it's a loss-less algorithm on text originating from english text it works better than available translators.

> imho this thought process is more of an analog process rather than a digital yes/no one.

On the quantum level, everything is digital. Analog is digital without raster-aligning after each operation.

> worse, just as frameworld said it happens on the subconscious level - not everyone can give detailed explanation (breakdown) of his/her own thought process.
> sure, you might be able to make a model for it (with effort) but…
> …might be easier to use another human language (just as what you asked in the 1st sentence).
> look at this: https://en.wikipedia.org/wiki/Javanese_language#Registers
> it descended from (indian) sanskrit but has evolved into its own.
> would be interesting if different aliens use different styles:

I can agree. Still, using a loss-less-transformation from english text has it's own benefits.

> drak uses krama - the most polite and formal one
> quarg/pug uses madya - the middle one
> hai/korath uses ngoko - the crude one disclaimer: not actually suggesting the bullet points above; those just meant to illustrate my point. i can only speak ngoko. the javanese alphabeths have been incorporated into the unicode btw.

That has a limit on how many language you have. You already show this by putting quark/pug and hai/korath into one category.


|| Also it needs somebody who speaks these languages to maintain the storyline and would look weird from a foreign speaker actually speaking these languages.

> Translating from a language to another requires understanding the meaning in language A, then finding a way to convey the same meaning in language B.

That's why I have already suggested a loss-less-transformation in my OP.


> It always means basically rewriting the whole text from scratch, and it NEVER involves any kind of weird trick as above.

In turn it requires WORK. Developing software is about doing initial work to reduce the total amount of work that will occur.


> Human languages are arbitrary binary associations between words and meanings, grammar is arbitrary, syntax is arbitrary, and also sayings are arbitrary, conventional logic will always fail horribly because a human language it's not anywhere near logic.

falsetrue XD.
90% is logic, 10% is applying the correct logic. Statistics can guess the 10% correctly in 80% cases (numbers are guesses - their value choice displays uncertainty).


> Say, I decide that "ship" is said "waisk".
> I also decide that in my language it's the singular form of the noun that needs a modifier to the word, and the modifier is 'kven.

The first, a computer can do (random: 1 to 26)+(random 1 to [0 to 26]) A car with navi, takes away choices on letters that would produce an invalid destination depending on your previous letters.
 * That would form syllables.

To start learning, just write a word wrong and think about why the firefox' auto-correction doesn't show you the correct word, even if you can guess it.

> And so on and so forth. Boom, I've made a language!
> Is it so hard?

Not now, but maintaining consistency will be. Especially when the initial devs have suddenly no time anymore.

> You can also make total masochist languages with ideograms like mandarin/japanese where each thing has a meaning alone and has or might have a different meaning if close to another thing, like the Tarot cards.

Reminds me of quotes "Yes, we can!" - each word has a meaning and the whole has it too.
"masochist" - has a meaning and a special meaning of "negative, self-damaging by making it artificially difficult" in your context.

> If I decide that the sound "wai" is codified by the symbol YY, and I decide that the sound "sk" is codified by the symbol W, when my aliens read "YYW" they say "waisk" with their voice and they mean "ships" in english.

Thought about it, but if you as foreign listener hear "wei", you write "wai", even if the alien writes "YY".
The story-line text is from your perspective, except some quotes of Alien text ("ÆL↑N . ł€Xł") in it.

> You can go on and make words for concepts that are difficult to translate directly, or don't have a word in english.

Masochist-Alien-Linguists do that :dancer:

> Giving them a language also allows "first contact" type missions where if you manage to understand their gibberish you can actually make better choices or progress better in their mission tree.
> Did you see the Kroath mission? That's a first contact, the Kroath you interact with don't speak in english.
> It is a stub so it is short, and has no effect on the game as a whole, but it's interesting.

Agree.


maybe it doesn't have to be a full-blown (read: proper) new alien language.
for the pilot, start with something as simple as alphabets substitution?

> make translator machine as a special outfit. without such outfit, the game should obscure simple english into an alien language.

––Meant–to–be–a–very–pessimistic–point–of–view––
* The super-computer would take 20 tons space like a Hyper-Drive and must be installed in your flag-ship, because if installed on any ship, it's an outfit that takes 0.2 tons on average in each ship of a big fleet.


> A translation outfit sounds nifty as hell - and would make an awesome plot-line. Imagine traveling toward the eastern rim of the galaxy, unable to communicate with anything until you run into some super-advanced empath-species with a translator, then traveling back toward human space brokering peace (or war, whatevs) among the races as you go.

Good idea. But I think the player should be able to construct it for himself too.

> Rather than implement one or more alien languages, though, it would probably hella simpler to use a block cipher and give each alien species a unique key that could be unlocked by the player and, through story-lines, given to other factions as well. Because I'm sure the unfettered Hai would just love to be able to communicate with the Korath.

Agree, but consider:
 * one key per language destroys mini-games of partially encrypting them.
 * I like some Yoda-speak transformator. "Yoda: Good, it smells".

**Comment by bobafetthotmail** [comment]
So, you want to make an automated alien language generator that is applying a bunch of cypher algorithms (at syntax level and at grammar level) to turn english in an "alien" language.

>"Your answer avoids a reference to the original context ('hw acc' for language | no rational rules you "know").
And I "know". Creating language is rational thinking for me. I can't understand why you don't "know" it."
Read this http://www.ecenglish.com/learnenglish/lessons/can-you-read

"I cnduo't bvleiee taht I culod aulaclty uesdtannrd waht I was rdnaieg. Unisg the icndeblire pweor of the hmuan mnid, aocdcrnig to rseecrah at Cmabrigde Uinervtisy, it dseno't mttaer in waht oderr the lterets in a wrod are, the olny irpoamtnt tihng is taht the frsit and lsat ltteer be in the rhgit pclae."

Can you read it? Yup (if you are fluent in english).

How do you read it? You can probably figure it out eventually because it's your field.
But even if you have no clue on how it works (like 99.9999% of people), it still works.

There are many other automatic abilities that allow us to understand broken sentences in languages we know.

>"In turn it requires WORK. Developing software is about doing initial work to reduce the total amount of work that will occur."

I was talking of translating between different real-life languages that exist already, you know, those that aren't all generated from a common base by applying cypher algorithms.

>"Not now, but maintaining consistency will be. Especially when the initial devs have suddenly no time anymore."

The language's grammar/syntax rules and vocabulary have to be written down somewhere in the source, it's obvious.
Since the level of communication needed in-game won't ever approach "ancient epic poetry" level, it will still remain relatively simple.

I meant that if something was never used (say none has written text talking of the very distant past), whoever has to write those parts can come up with the new language rules for it, or the same for words.

>"Reminds me of quotes "Yes, we can!" - each word has a meaning and the whole has it too."

No duh!

The point here is that a phonetic-alphabet-based system is error-tolerant enough to allow a good shot at successful communication even if you make major grammatical and syntactical mistakes.

Ideograms don't allow that. If you screw up the drawing, it's garbage, if you place the wrong ideograms close to each other, it's garbage.
Plus other random weirdness like the fact that many ideograms can be pronounced in 4 different ways, or that some ideograms have the same sound when read, so it's not easy to guess how to write down the spoken language.
Chinese people need powerful error-correction and word-guessing programs just to write something in a computer.
That's because their (written) language was designed from the start to be a semi-art restricted to elites, to show off how cool they were to plebs, more than being practical.

While most modern western languages are written forms of whatever was spoken by (the more literate) plebs, so the language was (relatively) flexible and error-tolerant.

Sure, you have constructs in all languages, like say "flat tire", but it's not crucial for successful communication, it's needed only to master the language.

Which is why Yoda-language is easy to make and perfectly readable with most western languages.
If you pull the same trick on ideograms like say chinese, there is a serious risk of turning it into unreadable garbage.

>"Thought about it, but if you as foreign listener hear "wei", you write "wai", even if the alien writes "YY"."

Yes, spoken alien language should be written with english phonetic alphabet, the written alien language can come into play if the player gets to read signs on the wall, or manuals or whatever.

>"Masochist-Alien-Linguists do that"

Making up words with complex, alien or otherwise hard-to-translate meanings gives that "alien" feel to the language. It also helps explaining the alien race's culture.
For example, an alien race where they use the same word for "children" and "daemon" what does tell you?

It's like spice though, a bit can turn the tide, a lot ruins the dish.

**Comment by frameworld** [comment]
I have few knowledge of ideograms.

> I cnduo't bvleiee taht I culod aulaclty uesdtannrd waht I was rdnaieg.

you can guess [2: couldn't from 't and cd du→ cu cud cudn couldn].
"uesdtannrd" is difficult because of so many syllables. 

It's all about constructing common syllables expected in the context (after I, verb that, reading/understand because it's related to what you do).

Try that with a word like "Donaudampfschiffskapitänsmützenhalterbrett"
Donau=river, dampf=steam, schiffs=ship's, kapitäns=captain's, mütze=hat, halter=holder, brett=board.

It works because there are only 5 vocals which can be spelled without help of these 5. Then there is some significant letter (d, L, t) that sticks out.
Creating about 25 possible syllables per word that have to fit into a context of about 300 raw possibilities. 
Often the most common is good enough which creates an average hit rate after likely 70 tries or less.
